src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,96 @@
+  });
+
+  it('Should be checked if the "checked" prop is true', () => {
+    render({ checked: true });
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).toBeChecked();
+  });
+
+  it('Should become checked when initially unchecked and user clicks', async () => {
+    const wrapper = renderAndGetWrapper({ checked: false });
+    await act(() => user.click(wrapper));
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).toBeChecked();
+  });
+
+  it('Should become unchecked when initially checked and user clicks', async () => {
+    const wrapper = renderAndGetWrapper({ checked: true });
+    await act(() => user.click(wrapper));
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).not.toBeChecked();
+  });
+
+  it('Should not change state when disabled and user clicks', async () => {
+    const wrapper = renderAndGetWrapper({ checked: false, disabled: true });
+    await act(() => user.click(wrapper));
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).not.toBeChecked();
+  });
+
+  it('Should not display checkbox when read-only', async () => {
+    render({ checked: false, readOnly: true });
+    const checkboxes = screen.queryAllByRole('checkbox');
+    expect(checkboxes).toHaveLength(0);
+  });
+
+  it('Should display label text', () => {
+    const label = 'Lorem ipsum';
+    render({ label });
+    expect(screen.getByText(label)).toBeDefined();
+  });
+
+  it('Should render checkbox with given checkboxId', () => {
+    const checkboxId = 'the-checkbox';
+    render({ checkboxId });
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox.id).toEqual(checkboxId);
+  });
+
+  it('Should render checkbox with given name', () => {
+    const name = 'the-checkbox';
+    render({ name });
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox.getAttribute('name')).toEqual(name);
+  });
+
+  it('Should become checked when "checked" prop changes from false to true', () => {
+    const { rerender } = render({ checked: false });
+    rerender(<Checkbox checked={true} />);
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).toBeChecked();
+  });
+
+  it('Should become unchecked when "checked" prop changes from true to false', () => {
+    const { rerender } = render({ checked: true });
+    rerender(<Checkbox checked={false} />);
+    const checkbox = screen.getByRole('checkbox');
+    expect(checkbox).not.toBeChecked();
+  });
+});
+
+const render = (props: Partial<CheckboxProps> = {}) => {
+  const allProps = { ...props };
+  return renderRtl(<Checkbox {...allProps} />);
+};
+
+const renderAndGetWrapper = (props: Partial<CheckboxProps> = {}): Element => {
+  const { container } = render(props);
+  const wrapper = container.querySelector('.wrapper');
+  expect(wrapper).not.toBeNull();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return wrapper!;
```

Comment:
This should be fixed instead of adding an eslint-disable statement - the eslint rules are there for a reason.

Reply:
But why does Eslint react on this at all? The line above says `expect(wrapper).not.toBeNull();`, so it should be impossible for this to be `null`. I could fix it with some `if` statement, but is that really necessary?

Reply:
There is no branching going on, and the code will not stop executing even if the wrapper is null because of the expect, so I think thats why?

Disabling rules as a one-off should be done only in very edge cases IMO (cant think of any good examples of when to do it at the moment). If we add them just to work around a minor annoyance, we will end up with a codebase littered with these statements (see Studio codebase @~1 year ago for an example), and they tend to get copied around to places they have no effect, and they are often left behind when the related code is cleaned up, and so on...

I think we should disable the rule altogether if we dont agree with it, or fix the code to comply with the rule.

Reply:
Ok, I thought the code would stop executing on `expect` errors. I replaced it `assert`.